examples/kg-chat/dependency_chatbot.py

```diff
@@ -0,0 +1,222 @@
+import typer
+from rich import print
+from rich.prompt import Prompt
+from pydantic import BaseSettings
+from dotenv import load_dotenv
+
+from langroid.agent.special.neo4j.neo4j_chat_agent import (
+    Neo4jChatAgent,
+    Neo4jChatAgentConfig,
+)
+from langroid.language_models.openai_gpt import OpenAIGPTConfig, OpenAIChatModel
+from langroid.utils.constants import NO_ANSWER
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.utils.configuration import set_global, Settings
+from langroid.utils.logging import setup_colored_logging
+from langroid.agent.tool_message import ToolMessage
+from langroid.agent.tools.google_search_tool import GoogleSearchTool
+
+from langroid.agent.task import Task
+from cypher_message import CONSTRUCT_DEPENDENCY_GRAPH
+
+
+app = typer.Typer()
+
+setup_colored_logging()
+
+
+class GetPackageInfo(ToolMessage):
+    request = "construct_dependency_graph"
+    purpose = f"""Get package <package_version>, <package_type>, and <package_name>.
+    For the <package_version>, obtain the recent version and should be a number. 
+    For the <package_type>, return if the package is PyPI or not.
+      Otherwise, return {NO_ANSWER}.
+    For the <package_name>, return the package name provided by the user.
+    ALL strings are in small letter. 
+    """
+    package_version: str
+    package_type: str
+    package_name: str
+
+
+class DependencyGraphAgent(Neo4jChatAgent):
+    package_name: str
+
+    def construct_dependency_graph(self, msg: GetPackageInfo) -> None:
+        check_db_exist = (
+            "MATCH (n) WHERE n.name = $name AND n.version = $version RETURN n LIMIT 1"
+        )
+        response = self.read_query(
+            check_db_exist, {"name": msg.package_name, "version": msg.package_version}
+        )
+        if "No records found" not in response:
+            self.config.database_created = True
+            return "Database Exists"
+        else:
+            construct_dependency_graph = CONSTRUCT_DEPENDENCY_GRAPH.format(
+                package_type=msg.package_type,
+                package_name=msg.package_name,
+                package_version=msg.package_version,
+            )
+            if self.write_query(construct_dependency_graph):
+                self.config.database_created = True
+                return "Database is created!"
+            else:
+                return f"""
+                    Database is not created!
+                    Seems the package {msg.package_name} is not found,
+                    """
+
+
+class CLIOptions(BaseSettings):
+    model: str = ""
+    fn_api: bool = False
+
+    class Config:
+        extra = "forbid"
+        env_prefix = ""
+
+
+def chat(opts: CLIOptions) -> None:
+    print(
+        """
+        [blue]Welcome to Dependency Analysis chatbot!
+        Enter x or q to quit at any point.
+        """
+    )
+
+    load_dotenv()
+
+    dependency_agent = DependencyGraphAgent(
+        config=Neo4jChatAgentConfig(
```

Comment:
Please add a way to get these from `.env` file similar to how various openai etc settings are obtained. I think Neo4jChatAgentConfig is mixing too many things-- credentials and ChatAgent related things like sys msg etc. Instead there should be a separate Neo4jSettings derived from Pyandic BaseSettings, with `config-prefix = `NEO4J` so that the settings are automatically obtained from `.env` file.

Reply:
The intention here was to follow the implementation of `SQLChatAgentConfig` https://github.com/langroid/langroid/blob/0e4fbad6be60e5662bc550cc3df1218007991572/langroid/agent/special/sql/sql_chat_agent.py#L66.

Reply:
Maybe you can just add `env_prefix  = "NEO4J_"` to this config class.